Add about page

arjunsalyan opened the pull request:
I am attaching a screenshot of the page, please let me know what more details need to be added, or if there is something that needs to be removed.
![Screenshot 2019-08-18 at 11 15 05 pm](https://user-images.githubusercontent.com/40331563/63228260-1c33d380-c20e-11e9-82ce-eaf4f58e296b.png)

arjunsalyan committed:
Add about page

arjunsalyan committed:
Add footer

arjunsalyan commented:
I have now deployed this branch on EC2: http://ec2-52-34-234-111.us-west-2.compute.amazonaws.com

The footer and about page can be seen live there.

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,57 @@
+
+
+
+{% endblock %}
```
Missing newline?

mojca commented on app/ports/templates/ports/layout.html:
```diff
@@ -49,6 +49,42 @@
+            </div>
+            <div class="col-lg-4">
+                Port Information was last updated at:<br>
+                <i class="footer-links">{{ port_info_updated_at|date:'Y-m-d H:i' }}</i><br><br>
```
Maybe add `UTC`?

mojca replied:
I would simply use `Port information was last updated at: <time> / <shortened shasum with link>` rather than using two separate sentences. The shortened shasum could use 8 characters, for example.

On the other hand we could add the timestamp of the last statistics submission and the timestamp of the last data fetching from buildbot as separate info.

arjunsalyan replied:
Thank you

> On the other hand we could add the timestamp of the last statistics submission and the timestamp of the last data fetching from buildbot as separate info.

We do not have the info as to when the build history was fetched last. We just store the timestamp of the builds. Shall we show the timestamp of the latest build we have in our database?

mojca commented on app/ports/templates/ports/layout.html:
```diff
@@ -49,6 +49,42 @@
+                <i class="footer-links">{{ port_info_updated_at|date:'Y-m-d H:i' }}</i><br><br>
+
+                Port Information is updated till the commit:
+                <i class="footer-links">{{ port_info_updated_till_commit }}</i>
```
What about creating a link to something like `https://github.com/macports/macports-ports/commits/<shasum>` ?

mojca reviewed (commented):
I really like the look & feel of the page footer, and I believe that one is ready for deployment. Some minor comments below. We can always easily add or remove links later depending on what we feel is important.

I have a few more suggestion for the about page which I will add later.

neverpanic commented:
Can we also include the information that we had on http://stats.macports.neverpanic.de/faq? The source is at https://trac.macports.org/browser/branches/gsoc11-statistics/stats-server/app/views/static/faq.html.erb.

arjunsalyan commented:
> http://stats.macports.neverpanic.de/faq

Thank you, that is very useful information to be included. I will add it.

mojca commented:
Adding the FAQ from the old site definitely makes sense. What could also make sense is to make a similar [Credits](http://stats.macports.neverpanic.de/credits) section. We could use some similar wording, but put the latest information, while keeping the rest as "historic" section. We could mention that the initial statistics collection has been implemented in Ruby on Rail as part of a 2011 GSOC project by ... and also mention that basically almost the same submission process (`mpstats`) is still used. The credits could also mention original authors of `portindex2json` etc.

arjunsalyan committed:
Add Stats FAQ Page

arjunsalyan committed:
Update the About page

arjunsalyan commented:
I have updated the footer, added the FAQ page and changed the credits section on about page.
The changes should be visible on EC2 within 15 minutes.
http://ec2-52-34-234-111.us-west-2.compute.amazonaws.com

I had no clue how to write the credits section, please let me know if I got things wrong.

mojca commented on app/ports/templates/ports/layout.html:
```diff
@@ -49,6 +49,47 @@
+            <div class="col-lg-4">
+                Port Information was last updated at:<br>
+                <i class="footer-links">{{ port_info_updated_at|date:'Y-m-d H:i' }} (UTC)</i><br>
+                <i><a class="footer-links" target="_blank" href="https://github.com/macports/macports-ports/commit/{{ port_info_updated_till_commit }}/">{{ port_info_updated_till_commit|slice:":8" }}</a></i><br><br>
```
I was thinking of this:
```suggestion
                <i><a class="footer-links" target="_blank" href="https://github.com/macports/macports-ports/commits/{{ port_info_updated_till_commit }}/">{{ port_info_updated_till_commit|slice:":8" }}</a></i><br><br>
```
or even better something along these lines
```suggestion
                <i><a class="footer-links" target="_blank" href="https://github.com/macports/macports-ports/commits/master?after={{ port_info_updated_till_commit }}+1">{{ port_info_updated_till_commit|slice:":8" }}</a></i><br><br>
```
But I cannot get the second one working as desired, it's always off by at least one; I guess it wasn't meant to be used like this.

And keys listed on https://developer.github.com/v3/repos/commits/ don't help either.

My idea was to show the list of the last few commits rather than just a single one.

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+                    <li><a href="https://github.com/macports/macports-webapp">Git Repository</a></li>
+                    <li><a href="https://github.com/macports/macports-webapp/tree/master/docs">Documentation</a></li>
+                    <li><a href="https://github.com/macports/macports-webapp/tree/master/docs/api_v1">API</a></li>
+                    <li><a href="https://docs.google.com/document/d/1TqNHm3NrtLDelWqY7P4pZyTMZDJrlsPymFZklzV2t24/edit">Feature Requests</a></li>
```
This link is soon about to "expire" and should be converted to some better design document. It can be left there if you want, but I assume it will be gone soon.

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+                    <li><a href="{% url 'stats_faq' %}">Statistics FAQ</a></li>
+                    <li><a href="https://github.com/macports/macports-webapp">Git Repository</a></li>
+                    <li><a href="https://github.com/macports/macports-webapp/tree/master/docs">Documentation</a></li>
+                    <li><a href="https://github.com/macports/macports-webapp/tree/master/docs/api_v1">API</a></li>
```
Not yet live, but you can leave it there.

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+                </li>
+                <li>Makes available all the collected data through API.</li>
+            </ol>
+            <p>The application is based on the <a href="https://www.djangoproject.com">Django Framework</a> and runs
```
Maybe the list of technologies could be moved to the bottom of credits?
Technologies being used certainly deserve to be credited, but with a much lower priority than the author of the app.

Then you could also avoid repeating the author and just mention the author once, at the very top of credits, as it is now.

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+    <h4>Credits</h4>
+    <hr>
+    <p>The MacPorts Webapp was written by <a href="https://github.com/arjunsalyan">Arjun Salyan</a> when MacPorts
+        participated in the Google Summer of Code 2019. The project was mentored by
```
```suggestion
        as a [Google Summer of Code 2019](https://trac.macports.org/wiki/SummerOfCode/2019#webapp) project, mentored by
```

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+    <br>
+    <h4>Credits</h4>
+    <hr>
+    <p>The MacPorts Webapp was written by <a href="https://github.com/arjunsalyan">Arjun Salyan</a> when MacPorts
```
```suggestion
    <p>The MacPorts Webapp was written by <a href="https://github.com/arjunsalyan">Arjun Salyan</a>
```

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+        The server is administered by <a href="https://github.com/neverpanic">Clemens Lang</a>.
+    </p>
+
+    <h5>Old Statistics Website</h5>
```
Maybe
```suggestion
    <h5>History</h5>
```
?

arjunsalyan replied:
Looks fine, thanks.

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+
+    <h5>Old Statistics Website</h5>
+    <p>
+	The old MacPorts statistics web application and the initial client were written by <a href="https://trac.macports.org/wiki/derek">Derek Ingrouville</a>
```
```suggestion
	The first MacPorts statistics web application and the initial client were written by <a href="https://trac.macports.org/wiki/derek">Derek Ingrouville</a>
```

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+    <h5>Old Statistics Website</h5>
+    <p>
+	The old MacPorts statistics web application and the initial client were written by <a href="https://trac.macports.org/wiki/derek">Derek Ingrouville</a>
+	when MacPorts participated in <a href="https://trac.macports.org/wiki/SummerOfCode2011">Google Summer of Code 2011</a>. Some documentation for the
```
```suggestion
	as a <a href="https://trac.macports.org/wiki/SummerOfCode2011">Google Summer of Code 2011</a> project. Some documentation for the
```

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+	The old MacPorts statistics web application and the initial client were written by <a href="https://trac.macports.org/wiki/derek">Derek Ingrouville</a>
+	when MacPorts participated in <a href="https://trac.macports.org/wiki/SummerOfCode2011">Google Summer of Code 2011</a>. Some documentation for the
+	source written in GSoC can be found in <a href="https://trac.macports.org/wiki/MacPortsStatisticsGSoC2011">the Wiki</a>. The source code for the
+	server is available in MacPorts' Subversion repository at <a href="https://trac.macports.org/browser/branches/gsoc11-statistics">/branches/gsoc11-statistics</a>.
```
There must be a better url than that. @neverpanic?

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+	server is available in MacPorts' Subversion repository at <a href="https://trac.macports.org/browser/branches/gsoc11-statistics">/branches/gsoc11-statistics</a>.
+    </p>
+    <br>
+    <h4>Dependencies</h4>
```
Maybe also technologies (Django etc.) could be mentioned at this place (moved from the start). Not necessarily under this section, but somewhere around here.

arjunsalyan replied:
Done, thanks.

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+        <br>
+            <p>
+	The client implementation as the <code>mpstats</code> port is the work of <a href="https://trac.macports.org/wiki/derek">Derek Ingrouville</a>,
+	<a href="https://trac.macports.org/wiki/jmr">Joshua Root</a> and <a href="https://trac.macports.org/wiki/cal">Clemens Lang</a>. The source of
```
We need to decide whether to cite developers under their trac pages or under their github handles (or both at all locations). Maybe not do a mixture of both (one citation with one and the other with another) on the same page. I don't have a strong preference for either.

arjunsalyan replied:
I could not find the right github handle for Derek, rest I have changed.

mojca commented on app/ports/templates/ports/about.html:
```diff
@@ -0,0 +1,73 @@
+            <p>
+	The client implementation as the <code>mpstats</code> port is the work of <a href="https://trac.macports.org/wiki/derek">Derek Ingrouville</a>,
+	<a href="https://trac.macports.org/wiki/jmr">Joshua Root</a> and <a href="https://trac.macports.org/wiki/cal">Clemens Lang</a>. The source of
+	the client is currently available at <a href="https://trac.macports.org/browser/trunk/dports/sysutils/mpstats">/trunk/dports/sysutils/mpstats</a>
```
Let's quote the github source instead.

mojca reviewed (commented):
Thanks a lot.
Some minor suggestions. Feel free to consider whatever you think is worth changing, then simply merge yourself. Any further changes can always be done later.

arjunsalyan committed:
Update the link to commits in footer

arjunsalyan committed:
Reduce number of category columns on index page for smaller screens

mojca commented on app/ports/templates/ports/stats_faq.html:
```diff
@@ -0,0 +1,76 @@
+            <strong>A unique identifier for your MacPorts installation</strong><br/>
+            This identifier is generated by <code>uuidgen</code> when you first
+            install the statistics service. We use this identifier to overwrite
+            statistics data you submitted earlier – we keep at most one set of
```
It is no longer true that we only keep at most one set per month. This "once per month" is mentioned again below.

arjunsalyan replied:
Thank you for picking this out.

mojca commented on app/ports/templates/ports/stats_faq.html:
```diff
@@ -0,0 +1,76 @@
+    <a name="opt-out"></a>
+    <h4>How Do I Stop Submitting Statistics?</h4>
+    <hr>
+    <p>Uninstalling the <a href="https://github.com/macports/macports-ports/tree/master/sysutils/mpstats">mpstats</a> port will prevent your system from
```
Now we could actually link to https://ports.macports.org/port/mpstats/ :)
There's another link above.

arjunsalyan replied:
I have updated the links.

mojca commented on app/ports/templates/ports/stats_faq.html:
```diff
@@ -0,0 +1,76 @@
+    <a name="opt-in"></a>
+    <h4>How Can I Submit Statistics?</h4>
+    <hr>
+    <p>To start submitting statistics, install the <a href="https://github.com/macports/macports-ports/tree/master/sysutils/mpstats">mpstats</a> port in
```
We could replace the link (see below) and probably write the explicit command `sudo port install mpstats` to make it even easier for users to do it.

arjunsalyan replied:
Thank you, I was about to add that.

mojca commented on app/ports/templates/ports/stats_faq.html:
```diff
@@ -0,0 +1,76 @@
+    <a name="collection-frequency"></a>
+    <h4>How Often Are Statistics Sent?</h4>
+    <hr>
+    <p>The statistics daemon runs once a week, overwriting the data your installation did submit earlier in the same
```
We are no longer overwriting data.

arjunsalyan replied:
Thank you.

mojca commented on app/ports/templates/ports/stats_faq.html:
```diff
@@ -0,0 +1,76 @@
+                <li>chosen variants</li>
+            </ul>
+        </li>
+    </ul>
```
We could add a sentence saying something like "You can check what gets sent by running `/opt/local/libexec/bin/mpstats show`". (Please check the path, I'm not behind my mac right now.)

arjunsalyan replied:
Yes, I was thinking of somehow including both the commands: `show/ submit`. Thank you.

mojca commented on app/ports/templates/ports/stats_faq.html:
```diff
@@ -0,0 +1,76 @@
+        <li>
+            <strong>Version numbers and system properties</strong><br/>
+            The statistics service collects and submits the versions of MacPorts,
+            OS X, GCC and Xcode installed on your system. It also tells us what
```
Did we review and update this list? Command line tools are missing. GCC could go at some point.

arjunsalyan replied:
I have updated the list, I have not removed gcc version for now.

arjunsalyan committed:
Update statistics FAQ

mojca commented:
Feel free to merge this whenever you think it is ready.

My only remaining feedback is that quick links on the about page look a bit strange, but we can also look at that one later.
